nexus/test-utils/src/lib.rs

```diff
@@ -681,9 +681,17 @@ impl<'a, N: NexusServer> ControlPlaneTestContextBuilder<'a, N> {
         sp_sim_config_file: Utf8PathBuf,
     ) {
         debug!(&self.logctx.log, "Starting Management Gateway");
-        let (mgs_config, sp_sim_config) =
+        let (mut mgs_config, sp_sim_config) =
             gateway_test_utils::setup::load_test_config(sp_sim_config_file);
 
+        if switch_location == SwitchLocation::Switch1 {
```

Comment:
Add a comment here?

I gather `start_gateway()` is invoked once for each switch, but we want to use the same config, which includes the location, so we need to fix up the second one here?